### DISCUSSION TITLE
Resolving deprecations when migrating 0.9.0 -> 1.2.0

### DISCUSSION THREAD

**Discussion by sgolubovic-interventure** [arrow-kt/arrow#3302] [Q&A] [answered]
I have a legacy project which is using Arrow 0.9.0 and I'm trying to migrate to 1.2.0. I managed to resolve all issues, except two.

Some parts of code have something like:

```kotlin
fun old(): Either<Throwable, Unit> = 
    Either.right(5)
        .map { log.info(it) }
        .unit()
```

which I replaced with

```kotlin
fun new(): Either<Throwable, Unit> = 
    Either.right(5)
        .map { log.info(it) }
        .map {}
```

but `map {}` seems like a workaround. Is there a better way to handle `void` methods?

---

Other situation is with `Either.cond`:

```kotlin
fun old(): Either<Throwable, Int> = 
    Either.cond(
        2 == 2,
        ifTrue = { 2 },
        ifFalse = { RuntimeException() }
    )
```

which I replaced with `Either.conditionally`

```kotlin
fun new(): Either<Throwable, Int> = 
    Either.conditionally(
        2 == 2,
        ifTrue = { 2 },
        ifFalse = { RuntimeException() }
    )
```

but `Either.conditionally` is deprecated and I couldn't find an equivalent. How to handle conditions in newer versions?

**Comment by kyay10** [comment, marked as answer]
`map {}` is probably the easiest way to signal your intent here.
For conditionally, you can just use `if (2 == 2) 2.right() else RuntimeException().left()`
The better way is to use `Raise`:
```kotlin
context(_: Raise<Throwable>)
fun ex1() { log.info(5) }
context(_: Raise<Throwable>)
fun ex2(): Int = if (2 == 2) 2 else raise(RuntimeException())
// or even

context(_: Raise<Throwable>)
fun ex2(): Int {
  ensure(2 == 2) { RuntimeException() }
  return 2
}
```
